inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp

```diff
@@ -164,6 +162,24 @@ void MKLDNNPlugin::MKLDNNInferRequest::InferImpl() {
                         pushInput<uint8_t>(input.first, input.second);
                     }
                     break;
+                case InferenceEngine::Precision::I64:
+                case InferenceEngine::Precision::U64: {
+                    // I64 and U64 is unsupported by mkldnn, so here we convert the blob and send I32
+                    iconv = InferenceEngine::make_shared_blob<std::int32_t>({InferenceEngine::Precision::I32,
+                                                                             input.second->getTensorDesc().getDims(),
+                                                                             input.second->getTensorDesc().getLayout()});
+                    convertedInputs.push_back(iconv);
+                    iconv->allocate();
+                    auto in = dynamic_cast<InferenceEngine::TBlob<std::int32_t> *>(iconv.get());
+                    if (in == nullptr)
+                        THROW_IE_EXCEPTION << "Cannot get TBlob";
+                    if (input.second->getTensorDesc().getPrecision() == InferenceEngine::Precision::I64)
+                        copyFrom<int64_t, int32_t>(input.second.get(), in->data());
+                    else
+                        copyFrom<uint64_t, int32_t>(input.second.get(), in->data());
+                    pushInput<int32_t>(input.first, iconv);
+                    }
+                    break;
```

Comment:
Could you please extract the code block above (lines 165 - 181) into separate method and reuse between other switch cases (U16, I16, BOOL).

Reply:
fixed